classes/galley/DAO.inc.php

```diff
@@ -0,0 +1,308 @@
+    }
+
+    public function get(int $id): ?Galley
+    {
+        return parent::get($id);
+    }
+
+    public function getByUrlPath(string $urlPath, Publication $publication): ?Galley
+    {
+        $row = DB::table($this->table)
+            ->where('publication_id', $publication->getId())
+            ->where('url_path', $urlPath)
+            ->first();
+        return $row ? $this->fromRow($row) : null;
+    }
+
+    /**
+     * Get the number of galleys matching the configured query
+     */
+    public function getCount(Collector $query): int
+    {
+        return $query
+            ->getQueryBuilder()
+            ->select('g.' . $this->primaryKeyColumn)
+            ->get()
+            ->count();
+    }
+
+    /**
+     * Get a list of ids matching the configured query
+     */
+    public function getIds(Collector $query): Collection
+    {
+        return $query
+            ->getQueryBuilder()
+            ->select('g.' . $this->primaryKeyColumn)
+            ->pluck('g.' . $this->primaryKeyColumn);
+    }
+
+    /**
+     * Get a collection of galleys matching the configured query
+     */
+    public function getMany(Collector $query): LazyCollection
+    {
+        $rows = $query
+            ->getQueryBuilder()
+            ->select(['g.*'])
+            ->get();
+
+        return LazyCollection::make(function () use ($rows) {
+            foreach ($rows as $row) {
+                yield $this->fromRow($row);
+            }
+        });
+    }
+
+    public function fromRow(object $row): Galley
+    {
+        $galley = parent::fromRow($row);
+
+        if (!empty($galley->getData('doiId'))) {
+            $galley->setData('doiObject', Repo::doi()->get($galley->getData('doiId')));
+        }
+
+        return $galley;
+    }
+
+    public function insert(Galley $galley): int
+    {
+        return parent::_insert($galley);
+    }
+
+    public function update(Galley $galley)
+    {
+        parent::_update($galley);
+    }
+
+    public function delete(Galley $galley)
+    {
+        parent::_delete($galley);
+    }
+
+    /** @copydoc RepresentationDAOInterface::getById() */
+    public function getById(int $galleyId, ?int $publicationId = null, ?int $contextId = null): ?Galley
+    {
+        return $this->get($galleyId);
+    }
+
+    /** @copydoc RepresentationDAOInterface::getByPublicationId() */
+    public function getByPublicationId(int $publicationId): array
+    {
+        return Repo::galley()->getMany(
+            Repo::galley()
+                ->getCollector()
+                ->filterByPublicationIds([$publicationId])
+        )->toArray();
+    }
+
+    /** @copydoc RepresentationDAOInterface::updateObject() */
+    public function updateObject(Representation $galley): void
+    {
+        $this->update($galley);
+    }
+
+    /**
+     * @copydoc PKPPubIdPluginDAO::pubIdExists()
+     */
+    public function pubIdExists($pubIdType, $pubId, $excludePubObjectId, $contextId)
```

Comment:
(would be nice to modernize this and some of the calls below -- but not an impediment to merging)

Reply:
Yes, the pub ID stuff is a real drag. But to remove it, we'd need to overhaul publication formats, submission files and issues. So I think it's going to be a while before we get there.